fabric-datafixer-v0/src/main/java/net/fabricmc/fabric/api/datafixer/FabricDataFixerUtils.java

```diff
@@ -0,0 +1,98 @@
+package net.fabricmc.fabric.api.datafixer;
+
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+
+import org.apache.commons.lang3.Validate;
+
+import com.google.common.collect.ImmutableMap;
+import com.mojang.datafixers.DataFix;
+import com.mojang.datafixers.DataFixer;
+import com.mojang.datafixers.DataFixerBuilder;
+import com.mojang.datafixers.schemas.Schema;
+
+import net.fabricmc.fabric.datafixer.mixin.util.SchemasAccessor;
+import net.fabricmc.fabric.impl.datafixer.FabricDataFixerInvoker;
+import net.minecraft.util.SystemUtil;
+import net.minecraft.util.crash.CrashException;
+
+/**
+ * This class is what registers and creates DataFixers for the game.
+ * <p><b>Please take extreme caution when using these tools as DataFixers directly interface with the world saves and may corrupt world saves.</b></p>
+ * @disclaimer <b>I, i509VCB and the Fabric Team do not take an responsibility for any worlds that are altered, destroyed, corrupted or damaged through the use of the {@link FabricDataFixerUtils} because of improper configuration, mispellings, incorrect syntax, etc.</b>
+ * @author i509
+ *
+ */
+public class FabricDataFixerUtils {
+    
+    private static final FabricDataFixerInvoker IMPL = FabricDataFixerInvoker.INSTANCE;
+    
+    static boolean LOCKED = false;
+    
+    public static final BiFunction<Integer,Schema,Schema> IDENTIFIER_NORMALIZE = SchemasAccessor.getIdentNormalize();
+    public static final BiFunction<Integer,Schema,Schema> EMPTY = SchemasAccessor.getEmpty();
+    
+    /**
+     * Creates a datafixer for a mod
+     * @param modid The modid of the mod
+     * @param runtimeDataVersion The current dataVersion of the mod
+     * @param builder A consumer in which to build the schema.
+     * @return The datafixer.
+     */
+    public static final DataFixer create(final String modid, final int runtimeDataVersion, final Consumer<DataFixerBuilder> builder) {
+        // Validation
+        Validate.notNull(modid, "modid cannot be null");
+        Validate.finite(runtimeDataVersion, "dataVersion must be finite");
```

Comment:
What... this `Validate.finite` is for checking double/float NaNs.

Also we tend to use guava preconditions over apache validate.